integrations/lib/plugindata/access_list.go

```diff
@@ -0,0 +1,52 @@
+// Copyright 2023 Gravitational, Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package plugindata
+
+import (
+	"strings"
+	"time"
+)
+
+// AccessListNotificationData represents generic plugin data required for access list notifications
+type AccessListNotificationData struct {
+	UserNotifications map[string]time.Time
+}
+
+// DecodeAccessListNotificationData deserializes a string map to PluginData struct.
+func DecodeAccessListNotificationData(dataMap map[string]string) (data AccessListNotificationData, err error) {
+	for user, notification := range dataMap {
+		if strings.HasPrefix(user, "un_") {
```

Comment:
This seems weird/hacky, what's `un_`?

Reply:
I don't disagree. `un_` is basically a prefix for "user notification." The way plugin data works, it has to be encoded into a `map[string]string`. I realize I could encode this as JSON instead of doing this, so I'll do this instead!